test/tests/jobs/RetryJobTest.cpp

```diff
@@ -334,6 +335,37 @@ struct RetryJobTest : tpunit::TestFixture {
         ASSERT_EQUAL(difftime(nextRunTime, createdTime), 3600);
     }
 
+    // Retry a job with a repeat that uses the custom "START OF HOUR" modifier
+    void hasRepeatStartOfHour() {
+        uint64_t now = STimeNow();
+
```

Comment:
Does this test break if the clock ticks past midnight right here? I think it does, we will have the wrong day for the final ASSERT_EQUAL.

Reply:
I set `command["nextRun"] = now;` below to try to avoid this. Is there another way to avoid it being flaky?

Reply:
This is what gets saved in the DB for `nextRun` in `CreateJob`, right?
https://github.com/Expensify/Bedrock/blob/2477dd1c79faabed66660126e322521d56aa895a/plugins/Jobs.cpp#L605C29-L605C41

Which is set here:
https://github.com/Expensify/Bedrock/blob/2477dd1c79faabed66660126e322521d56aa895a/plugins/Jobs.cpp#L505-L506

So you want to set `job["firstRun"]`, correct?

But `lastRun` is set here in `GetJob`:
https://github.com/Expensify/Bedrock/blob/2477dd1c79faabed66660126e322521d56aa895a/plugins/Jobs.cpp#L820

And then `RetryJob` will use that valuer of `lastRun` to compute the next run time, I think. Correct me if that's wrong.

Reply:
Yeah it looks like if we use `STARTED` in the repeat, it will use last run, so I think we want to use `SCHEDULED` instead, which gets set to `nextRun` here: https://github.com/Expensify/Bedrock/blob/2477dd1c79faabed66660126e322521d56aa895a/plugins/Jobs.cpp#L1127-L1129